splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceDateFormatter.java

```diff
@@ -14,29 +14,88 @@
 
 package com.splicemachine.derby.utils;
 
+import com.splicemachine.db.iapi.reference.SQLState;
+import java.sql.SQLException;
+import static java.lang.String.format;
+
 public class SpliceDateFormatter {
 
     private String format = null;
-    private java.time.format.DateTimeFormatter formatter;
+    private java.time.format.DateTimeFormatter formatter = null;
 
     public SpliceDateFormatter(String format) {
-        this.format = format != null ? format : "yyyy-M-d";
-        
-        if (format != null) {
+
+        // The default external format of "yyyy-MM-dd" may be used
+        // in error messages, but the corresponding internally used format
+        // is "yyyy-M-d".
+        if (format == null) {
+            format = "yyyy-M-d";
+            this.format = "yyyy-MM-dd";
+        }
+        else {
+            this.format = format;
+
+            // The valid format characters of DateTimeFormatter documented at the following link
+            // can be used to identify separator characters like '-' or '/' :
+            // See https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html
+
+            // Any non-format character can be a separator character.
+            // This pattern matches one or more non-format characters.
+            String
```

Comment:
Why is the above comment not part of javadoc of the class?

Reply:
This is an implementation detail, which should be an internal comment.  I'll add some javadoc comments referencing splice date formats.